src/pynlo/light/PulseBase.py

```diff
@@ -636,13 +636,15 @@ def add_noise(self, noise_type='sqrt_N_freq'):
         nothing
         """
         
-        # This is all to get the number of photons in each frequency bin:
-        size_of_bins = np.gradient(self.W_THz)
-        energy_per_bin = np.abs(self.AW)**2/size_of_bins * 1e-12
-    
-        h = 6.62607004e-34
-        photon_energy = h * self.W_THz/(2*np.pi) * 1e12
-        photons_per_bin = energy_per_bin/photon_energy
+        # This is all to get the number of photons/second in each frequency bin:
+        size_of_bins = self.dF_mks                          # Bin width in [Hz]
```

Comment:
since the bin size has been changed to Hz, rather than THz, the factor of 1e12 should be removed from line 661